### PR TITLE
[SDP-1146] Delete Tenant

### PR DESCRIPTION
**Pull request by ziyliu**
### What
- Add ability to "delete" tenant when tenant is in the `DEACTIVATED` status
- Filter out tenants that have been "deleted" in the `GET` endpoints

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

**Comment by reecexlm** [comment]
we should probably check into why k8s are in crash loop. stellar-disbursement-platform-backend-pr272-7777d754f8-c2mzz      0/1     CrashLoopBackOff

File: stellar-multitenant/pkg/tenant/manager.go [hunk 1]
```diff
@@ -76,9 +77,7 @@ var selectQuery string = `
 func (m *Manager) GetAllTenants(ctx context.Context, queryParams *QueryParams) ([]Tenant, error) {
 	if queryParams == nil {
 		queryParams = &QueryParams{
-			Filters: map[FilterKey]interface{}{
-				FilterKeyOutStatus: DeactivatedTenantStatus,
-			},
+			Filters:   excludeInactiveTenantsFilters(),
```
Comment (marwen-abid) [inline stellar-multitenant/pkg/tenant/manager.go:80]: Nice!

**Review by marwen-abid — APPROVED** [review]
LGTM!

File: stellar-multitenant/internal/httphandler/tenants_handler.go [hunk 2]
```diff
@@ -162,6 +167,63 @@ func (t TenantsHandler) Patch(w http.ResponseWriter, r *http.Request) {
 	httpjson.RenderStatus(w, http.StatusOK, tnt, httpjson.JSON)
 }
 
+func (t TenantsHandler) Delete(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	tenantID := chi.URLParam(r, "id")
+
+	tnt, err := t.Manager.GetTenant(ctx, &tenant.QueryParams{
+		Filters: map[tenant.FilterKey]interface{}{tenant.FilterKeyID: tenantID},
+	})
```
[the remark this answers is not in the document]
Comment (ziyliu) [reply stellar-multitenant/internal/httphandler/tenants_handler.go:178]: addressed here! #[7b969dd](https://github.com/stellar/stellar-disbursement-platform-backend/pull/272/commits/7b969dd8a21dcd9c8c14fd03d82b0d7364629702)

**Pull request merged**